File: src/main/java/ru/chernov/urlshortener/mapper/UserMapper.java
package ru.chernov.urlshortener.mapper;

import org.mapstruct.Mapper;
import ru.chernov.urlshortener.dto.user.UserResponse;
import ru.chernov.urlshortener.entity.user.User;


@Mapper
public interface UserMapper {

    UserResponse toResponse(User user);

}

File: src/main/kotlin/ru/chernov/urlshortener/mapper/UserMapper.kt
package ru.chernov.urlshortener.mapper

import org.springframework.stereotype.Component
import ru.chernov.urlshortener.dto.user.UserResponse
import ru.chernov.urlshortener.entity.user.User

// TODO: add mapstruct
@Component
class UserMapper {

    fun toResponse(user: User): UserResponse {
        return UserResponse(user.id, user.username, user.token)
    }

}